src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AccessController.cs

```diff
@@ -192,16 +211,30 @@ public class AccessController : Controller
                 });
             }
 
-            if (application.SkipConsent)
+            var openIdSettings = await _openIdService.GetOpenIdSettingsAsync();
+            var isAuthorizationCodeFlow = (request.IsAuthorizationCodeFlow() && openIdSettings.AllowAuthorizationCodeFlow && application != null && application.AllowAuthorizationCodeFlow);
```

Comment:
Same remark, this must be reworked to avoid unnecessary checks and return a meaningful error to the caller.